Linker Error With octomapWrapper.cpp.o - Undefined Reference to createChild()

I'm having an issue regarding using make as instructed in http://drake.mit.edu/from_source.html and http://drake.mit.edu/ubuntu_trusty.html

I've also set up my environment variables in my ~/.bashrc file as such:

export CC=gcc-4.9 CXX=g++-4.9 FC=gfortran-4.9 F77=gfortran-4.9

The problem is experienced when I try to compile in my build folder in the drake-distro directory, of which I got from "git clone git@github.com:RobotLocomotion/drake.git drake-distro", after using "cmake .." and  "make". Upon using make, it seems to fails when it attempts to link a certain shared module. This is as illustrated in the figure below:

![error_msg](https://cloud.githubusercontent.com/assets/14330718/19414175/51f59912-933c-11e6-8a5a-03171f550f0a.png)

These are other details that relate to my computer environment:
Distributor ID: Ubuntu
Description:    Ubuntu 14.04.5 LTS
Release:    14.04
Codename:   trusty

MATLAB 9.1.0.441655 (R2016b)

java version "1.8.0_101"
Java(TM) SE Runtime Environment (build 1.8.0_101-b13)
Java HotSpot(TM) 64-Bit Server VM (build 25.101-b13, mixed mode)

I wait eagerly for a response. Thank you for your time.

Here is the error I'm observing when building Drake using `catkin` on Ubuntu 14.04 + ROS Indigo:

```
In file included from /home/liang/dev/drake_catkin_workspace/src/drake/externals/fcl/src/geometry/octree/octree.cpp:38:
/home/liang/dev/drake_catkin_workspace/src/drake/externals/fcl/include/fcl/geometry/octree/octree-inl.h:179:16: error: no member named 'getNodeChild' in 'octomap::OcTree'
  return tree->getNodeChild(node, childIdx);
         ~~~~  ^
/home/liang/dev/drake_catkin_workspace/src/drake/externals/fcl/include/fcl/geometry/octree/octree-inl.h:191:16: error: no member named 'getNodeChild' in 'octomap::OcTree'
  return tree->getNodeChild(node, childIdx);
         ~~~~  ^
/home/liang/dev/drake_catkin_workspace/src/drake/externals/fcl/include/fcl/geometry/octree/octree-inl.h:203:16: error: no member named 'nodeChildExists' in 'octomap::OcTree'
  return tree->nodeChildExists(node, childIdx);
         ~~~~  ^
/home/liang/dev/drake_catkin_workspace/src/drake/externals/fcl/include/fcl/geometry/octree/octree-inl.h:214:16: error: no member named 'nodeHasChildren' in 'octomap::OcTree'
  return tree->nodeHasChildren(node);
         ~~~~  ^
```
When building `drake/externals/fcl`, the build system is apparently finding ROS Indigo's older version of `octomap` rather than Drake's newer version. ROS' version of `octomap` is part of Debian package `ros-indigo-octomap`:

```
$ dpkg -l | grep octomap
ii  ros-indigo-octomap                                    1.6.9-0trusty-20160321-174526-0700                  amd64        The OctoMap library implements a 3D occupancy grid mapping approach, providing data structures and mapping algorithms in C++. The map implementation is based on an octree. See http://octomap.github.io for details.
ii  ros-indigo-octomap-msgs                               0.3.3-0trusty-20160627-171624-0700                  amd64        This package provides messages and serializations / conversion for the OctoMap library.
```

Drake's version of `octomap` is in `drake-distro/externals/octomap`.

FCL's build system is searching for `octomap` using the [following logic](https://github.com/flexible-collision-library/fcl/blob/06d48b3b6f3605b8caf119d5208d8156eb64fe0d/CMakeLists.txt#L143):

```
  find_package(octomap QUIET)
  # octomap-config.cmake may not define OCTOMAP_VERSION so fall back to
  # pkgconfig
  if(NOT OCTOMAP_VERSION AND PKG_CONFIG_FOUND)
      pkg_check_modules(OCTOMAP QUIET octomap)
  endif()
```

I attempted to prioritize Drake's version of FCL and Octomap by using the `CMAKE_CXX_FLAGS` flag. For example:

```
$ cd ~/dev/drake_catkin_workspace
$ catkin config --cmake-args -DCMAKE_BUILD_TYPE:STRING=Debug -DWITH_DREAL=FALSE -DDISABLE_MATLAB=TRUE -DCMAKE_CXX_FLAGS="-I`pwd`/src/drake/externals/octomap/octomap/include -I`pwd`/install/include"
```

Unfortunately, I couldn't get it to work after trying various combination of paths in the set [src/, build/, install/]. For example, one error was:

```
In file included from /home/liang/dev/drake_catkin_workspace/src/drake/drake/multibody/collision/test/fcl_model_test.cc:3:
In file included from /home/liang/dev/drake_catkin_workspace/src/drake/../../install/include/fcl/fcl.h:62:
In file included from /home/liang/dev/drake_catkin_workspace/src/drake/../../install/include/fcl/geometry/octree/octree.h:145:
/home/liang/dev/drake_catkin_workspace/src/drake/../../install/include/fcl/geometry/octree/octree-inl.h:178:5: error: function-like macro 'OCTOMAP_VERSION_AT_LEAST' is not defined
#if OCTOMAP_VERSION_AT_LEAST(1,8,0)
```

--- a/drake/doc/from_source_ros_indigo.rst
+++ b/drake/doc/from_source_ros_indigo.rst
@@ -10,8 +10,12 @@ Step 1: Install Prerequisites
 =============================
 
 Install `Ubuntu 14.04.4 LTS (Trusty Tahr) <http://releases.ubuntu.com/14.04/>`_
-and `ROS Indigo <http://wiki.ros.org/indigo>`_. We recommend installing the
-"desktop-full" version of ROS to get its full feature set.
+and `ROS Indigo <http://wiki.ros.org/indigo>`_. We recommend initially
+installing the "desktop-full" version of ROS, but then uninstalling the
+following two packages due to incompatibility with Drake (see
+`issue #3814 <https://github.com/RobotLocomotion/drake/issues/3814>`_)::
+
+    sudo apt-get remove ros-indigo-octomap ros-indigo-fcl
 
 Add your public SSH key to your github.com account by following
 `these instructions <https://help.github.com/articles/adding-a-new-ssh-key-to-your-github-account/>`_. This is necessary because the

